src/lisp/kernel/lsp/pprint.lisp

```diff
@@ -705,7 +710,14 @@
 ;;;; Utilities.
 
 (defun pprint-pop-helper (object count stream &aux code)
-  (cond ((not (listp object))
+  (cond ((or (not (listp object))
+             (and (plusp count)
+                  (consp (cdr object))
+                  (null (cddr object))
+                  (or (eq (car object) 'core:quasiquote)
+                      (and (getf core:*quasiquote* (target-stream stream))
```

Comment:
this getf is showing up enough that it might be worthwhile defining a `core:` function for it - `printing-quasiquote-p` or something - to hide the details of what's in `core:*quasiquote*` a bit. Though format-pprint up there would still have to bind it.

Reply:
I was going to try do a similar treatment for circle detection. Basically, make the 2-phase circle detection more robust in case of nesting. That is more complex and I need to think about it a bit more. So maybe wait on generalizing or abstracting this idea until then?

Reply:
Works for me.

Reply:
Cool. Thanks.